java-checks/src/main/java/org/sonar/java/checks/RedundantThrowsDeclarationCheck.java

```diff
@@ -216,14 +219,47 @@ private static boolean emptyBody(MethodTree methodTree) {
     if (block != null) {
       MethodInvocationVisitor visitor = new MethodInvocationVisitor();
       block.accept(visitor);
-      return visitor.thrownExceptions();
+      Set<Type> thrownExceptions = visitor.thrownExceptions();
+
+      if (thrownExceptions != null && methodTree.is(Tree.Kind.CONSTRUCTOR) && !visitor.visitedSuper) {
+        Optional<Symbol.MethodSymbol> constructor = getImplicitlyCalledConstructor(methodTree);
+        constructor.ifPresent(c -> thrownExceptions.addAll(c.thrownTypes()));
+      }
+
+      return thrownExceptions;
     }
     return null;
   }
 
+  private static Optional<Symbol.MethodSymbol> getImplicitlyCalledConstructor(MethodTree methodTree) {
+    Tree parent = methodTree.parent();
+    if (parent != null && parent.is(Tree.Kind.CLASS)) {
+      Type superType = ((ClassTree) parent).symbol().superClass();
+      if (superType != null) {
+        List<Symbol.MethodSymbol> candidates = superType.symbol().memberSymbols().stream()
+          .filter(RedundantThrowsDeclarationCheck::isDefaultConstructor)
+          .map(Symbol.MethodSymbol.class::cast)
+          .collect(Collectors.toList());
+        if(candidates.size() == 1) {
+          return Optional.of(candidates.get(0));
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  private static boolean isDefaultConstructor(Symbol symbol) {
+    if (symbol.isMethodSymbol()) {
+      MethodTree methodTree = ((Symbol.MethodSymbol)symbol).declaration();
```

Comment:
when the class is not in the same file, you don't have access to the constructor. Check for a symbol called `"<init>"` without parameters. Don't restrain yourself to what we can do with the syntax only.